generators/app/templates/src/StarterKit/Startup/Startup.cs

```diff
@@ -1,114 +1,178 @@
-using Swashbuckle.AspNetCore.Swagger;
-using System;
-using System.IO;
-using System.Reflection;
 
-namespace StarterKit
+namespace StarterKit.Startup
 {
   public class Startup
   {
-    public Startup(IConfiguration configuration, IHostingEnvironment env)
+    public Startup(IConfiguration configuration, IHostEnvironment env)
     {
+      ApplicationBasePath = env.ContentRootPath;
       Configuration = configuration;
+      Environment = env;
     }
 
-    public IConfiguration Configuration { get; private set; }
-    public string ApplicationBasePath { get; private set; }
+    public IConfiguration Configuration { get; }
+    public string ApplicationBasePath { get; }
+    public IHostEnvironment Environment { get; }
+    private string XmlCommentsPath
+    {
+      get
+      {
+        var fileName = typeof(Startup).GetTypeInfo().Assembly.GetName().Name + ".xml";
+        return Path.Combine(ApplicationBasePath, fileName);
+      }
+    }
 
-    public void ConfigureServices(IServiceCollection services)
+    public virtual void ConfigureServices(IServiceCollection services)
     {
+      #region Read settings
+
       // Check out ExampleController to find out how these configs are injected into other classes
-      AppSettings.RegisterConfiguration(services, Configuration.GetSection("AppSettings"));
+      var appSettingsSection = Configuration.GetSection("AppSettings");
+      var appSettings = appSettingsSection.Get<AppSettings>();
+      services.AddSingleton(Configuration).Configure<AppSettings>(appSettingsSection);
 
-      var appSettings = services.BuildServiceProvider().GetService<IOptions<AppSettings>>().Value;
+      #endregion
+
+      #region Add Correlation and application services
+
+      services.AddCorrelation(options =>
+      {
+        options.CorrelationHeaderRequired = !Environment.IsDevelopment();
+      });
 
       services.AddApplicationServices(opt =>
       {
         opt.ApplicationId = appSettings.ApplicationId;
         opt.ApplicationName = appSettings.AppName;
       });
 
-      services.AddCorrelation(options => { options.CorrelationHeaderRequired = true; });
+      #endregion
+
+      #region Logging
 
       services.AddLoggingEngine();
 
+      #endregion
+
       //--dataaccess-startupServices--
 
-      services.AddMvc()
-          .AddJsonOptions(options =>
-          {
-            options.SerializerSettings.ReferenceLoopHandling = Newtonsoft.Json.ReferenceLoopHandling.Ignore;
-            options.SerializerSettings.Converters.Add(new Newtonsoft.Json.Converters.StringEnumConverter());
-          })
-          .AddApiExtensions(null, options =>
-          {
-            options.DisableVersioning = true;
-          });
+      #region Add routing and versioning
+
+      services
+        .AddRouting(options =>
+        {
+          options.LowercaseUrls = true;
+          options.LowercaseQueryStrings = true;
+        })
+        .AddControllers()
+        .AddNewtonsoftJson(options =>
```

Comment:
Mss de optie toevoegen in comments om de nieuwe Json serializer te gebruiken

`.AddJsonOptions(options =>
                {
                    options.JsonSerializerOptions.PropertyNameCaseInsensitive = true;
                    options.JsonSerializerOptions.Converters.Add(new JsonStringEnumConverter());
                    options.JsonSerializerOptions.Converters.Add(new AbstractPropertyJsonConverter());
                });`

De AbstractPeropertyJsonConvert zorgt voor ondersteuning van serializatie met inheritance

`public class AbstractPropertyJsonConverter : JsonConverter<object>
    {
        public override bool CanConvert(Type typeToConvert)
        {
            return typeToConvert != null ? typeToConvert.IsAbstract : false;
        }

        public override object Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
        {
            throw new NotImplementedException();
        }

        public override void Write(Utf8JsonWriter writer, object value, JsonSerializerOptions options)
        {
            JsonSerializer.Serialize(writer, value, value?.GetType(), options);
        }
    }`